### PR TITLE
Add budget allocated and projected fields to projects.list, projects.info

### PR DESCRIPTION
**Pull request by joaoinacio**
Implementation PR: https://github.com/teamleadercrm/core/pull/10582
E2E Tests: https://github.com/teamleadercrm/api-tests/pull/754

### CODE REVIEW

**Review by ArnaudWeyts — APPROVED** [review]

**Review by adognameddog — APPROVED** [review]

**Review by psrpinto — APPROVED** [review]

**Review by jenssegers — APPROVED** [review]

File: apiary.apib [hunk 1]
```diff
@@ -3055,6 +3057,8 @@ Get details for a single project.
                 + `provided` (Money)
                 + `spent` (Money)
                 + `remaining` (Money) - The actual budget remaining, not taking the planned work into account.
+                + `allocated` (Money, optional) - The estimated budget for planned work that is still remaining.
```
Comment (jenssegers) [inline apiary.apib:3060]: I find these descriptions a bit difficult to understand, can we check with the PO for a better description?

**Review by ketlane — APPROVED** [review]

File: apiary.apib [hunk 2]
```diff
@@ -3055,6 +3057,8 @@ Get details for a single project.
...
+                + `projected` (Money, optional) - The estimated budget at completion.
```
Comment (ketlane) [inline apiary.apib:3061]: Q: Are `allocated` and `projected` really optional? Why are they different to `provided`, `spent` and `remaining`?

**Review by ketlane — APPROVED** [review]

File: apiary.apib (hunk 2 above)
Comment (joaoinacio) [reply apiary.apib:3061]: > Q: Are `allocated` and `projected` really optional? Why are they different to `provided`, `spent` and `remaining`?

First 3 are always specified, allocated and projected may or may not exist yet (`null`)

Comment (ketlane) [reply apiary.apib:3061]: But that should be marked with `nullable`, not `optional`, no? They will always be there, but they may be null.

**Comment by joaoinacio** [comment]
Closing in favor of https://github.com/teamleadercrm/api/pull/454

**Pull request closed without merging**